TypewiseAlert.Test/TypewiseAlertTest.cs

```diff
@@ -1,15 +1,195 @@
+using Moq;
 using System;
+using System.Collections.Generic;
 using Xunit;
 
 namespace TypewiseAlert.Test
 {
-  public class TypewiseAlertTest
-  {
-    [Fact]
-    public void InfersBreachAsPerLimits()
+    public class TypewiseAlertTest
     {
-      Assert.True(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+        public static IEnumerable<object[]> CheckAndAlertForNormalBreach =>
+        new List<object[]>
+        {
+            new object[] { AlertTarget.TO_CONSOLE,
+                new TypewiseAlert.BatteryCharacter() { coolingType = CoolingType.PASSIVE_COOLING,brand="BOSCH"}, 12 },
+            new object[] { AlertTarget.TO_CONTROLLER,
+                new TypewiseAlert.BatteryCharacter() { coolingType = CoolingType.MED_ACTIVE_COOLING,brand="BOSCH"}, 12 },
+            new object[] { AlertTarget.TO_EMAIL,
+                new TypewiseAlert.BatteryCharacter() { coolingType = CoolingType.HI_ACTIVE_COOLING,brand="BOSCH"}, 12 },
+        };
+
+
+        [Fact(DisplayName = "Metadata utility test")]
+        public void metadataUtilityTest()
+        {
+            var instance = MetaDataUtility.CreateInstanceWithInterfaceAndAttribute("TypewiseAlert", "TypewiseAlert", typeof(IAlertRaiser),
+                "TO_FAKE_EMAIL");
+            Assert.NotNull(instance);
+        }
+
+        [Fact(DisplayName = "Infer breach for passive cooling - low")]
+        public void classifyTemperatureBreachForPassiveCoolingTest()
+        {
+            Assert.True(TypewiseAlert.ClassifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -1) ==
+                BreachType.TOO_LOW);
+        }
+
+        [Fact(DisplayName = "Infer breach for medium active cooling - normal")]
+        public void classifyTemperatureBreachForMediumCoolingTest()
+        {
+            Assert.True(TypewiseAlert.ClassifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 12) ==
+                BreachType.NORMAL);
+        }
+
+        [Fact(DisplayName = "Infer breach for High active cooling - high")]
+        public void classifyTemperatureBreachForHighCoolingTest()
+        {
+            Assert.True(TypewiseAlert.ClassifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 46) ==
+                BreachType.TOO_HIGH);
+        }
+
+        [Theory(DisplayName = "When breach infered is TOO_LOW and TOO_HIGH and FAKE CONSOLE alert triggered")]
+        [InlineData(BreachType.TOO_LOW)]
+        [InlineData(BreachType.TOO_HIGH)]
+        public void fakeConsoleCheckAndAlertTest(BreachType breachType)
+        {
+            //Arrange
+            var fakeConsoleNotifier = new FakeConsoleNotifier();
```

Comment:
who is the CodeUnderTest ....expected to call CheckAndAlert

Reply:
Correct sir check and alert should be code under test .. I thought call check and alert but since we cannot change the enum types for test purpose. So I did not call it even knowing it.
Can you suggest me how to separate create instance from check and alert ..?